### PR TITLE
Master relational model fge

### PR DESCRIPTION
**Pull request by FrancoisGe**

### CODE REVIEW

File: addons/web/static/src/views/relational_model/static_list.js [hunk 1]
```diff
@@ -461,8 +461,16 @@ export class StaticList extends DataPoint {
...
+        await this.model._updateConfig(this.config, { limit, offset, orderBy }, { noReload: true });
+        const currentIds = this._currentIds.slice(offset, offset + limit);
+        const resIds = currentIds.filter((id) => typeof id === "number" && !this._cache[id]);
+        if (resIds.length) {
+            const records = await this.model._loadRecords({ ...this.config, resIds });
+            for (const record of records) {
+                this._createRecordDatapoint(record);
+            }
+        }
+        this.records = currentIds.map((id) => this._cache[id]);
```
Comment (FrancoisGe) [inline addons/web/static/src/views/relational_model/static_list.js:476]: at the end         
await this.model._updateConfig(this.config, { limit, offset, orderBy }, { noReload: true });

File: addons/web/static/src/views/relational_model/static_list.js [hunk 2]
```diff
@@ -461,8 +461,16 @@ export class StaticList extends DataPoint {
     }
 
     async _load({ limit, offset, orderBy }) {
-        const records = await this.model._updateConfig(this.config, { limit, offset, orderBy });
-        this.records = records.map((r) => this._createRecordDatapoint(r));
+        await this.model._updateConfig(this.config, { limit, offset, orderBy }, { noReload: true });
+        const currentIds = this._currentIds.slice(offset, offset + limit);
+        const resIds = currentIds.filter((id) => typeof id === "number" && !this._cache[id]);
```
Comment (FrancoisGe) [inline addons/web/static/src/views/relational_model/static_list.js:469]: FIX me record in cache not yet loaded

File: addons/web/static/tests/views/fields/one2many_field_tests.js [hunk 3]
```diff
@@ -8428,17 +8434,28 @@ QUnit.module("Fields", (hooks) => {
...
-                if (args.method === "onchange") {
-                    return Promise.resolve({
+            async mockRPC(route, args) {
+                if (args.method === "onchange2") {
+                    return {
                         value: {
                             turtles: [
-                                [5],
+                                [2, 2],
+                                [2, 3],
```
Comment (FrancoisGe) [inline addons/web/static/tests/views/fields/one2many_field_tests.js]: remove

File: addons/web/static/tests/views/fields/one2many_field_tests.js [hunk 4]
```diff
@@ -8428,17 +8434,28 @@ QUnit.module("Fields", (hooks) => {
...
+                                [
+                                    4,
+                                    1,
+                                    { id: 1, turtle_int: 0, turtle_foo: "yop", partner_ids: [] },
+                                ],
+                                [
+                                    4,
+                                    3,
+                                    { id: 3, turtle_int: 21, turtle_foo: "kawa", partner_ids: [] },
+                                ],
```
Comment (FrancoisGe) [inline addons/web/static/tests/views/fields/one2many_field_tests.js]: remove

File: addons/web/static/tests/views/fields/one2many_field_tests.js [hunk 5]
```diff
@@ -8428,17 +8434,28 @@ QUnit.module("Fields", (hooks) => {
...
                                 [1, 1, { turtle_foo: "from onchange id2", partner_ids: [[5]] }],
                                 [1, 3, { turtle_foo: "from onchange id3", partner_ids: [[5]] }],
```
Comment (FrancoisGe) [inline addons/web/static/tests/views/fields/one2many_field_tests.js]: remove partner_ids

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**